lib/src/native/realm_core.dart

```diff
@@ -1672,15 +1672,26 @@ class LastError {
   }
 }
 
-abstract class HandleBase<T extends NativeType> {
+void _traceFinalization(Object o) {
+  Realm.logger.log(RealmLogLevel.trace, 'Finalizing: $o');
+}
+
+final _debugFinalizer = Finalizer<Object>(_traceFinalization);
```

Comment:
This is just for tracing what happens on the dart side. It is automatically dropped in release mode, since it runs as part of an assert, and all it does is call `Realm.logger.log(RealmLogLevel.trace, 'Finalizing: $o');`.

Reply:
Reworked as discussed on round-table meeting